xblockutils/studio_editable.py

```diff
@@ -0,0 +1,295 @@
+                for entry in list_values:
+                    entry["value"] = json.dumps(entry["value"])
+            info['list_values'] = list_values
+            info['has_list_values'] = True
+        return info
+
+    @XBlock.json_handler
+    def submit_studio_edits(self, data, suffix=''):
+        """
+        AJAX handler for studio_view() Save button
+        """
+        values = {}  # dict of new field values we are updating
+        to_reset = []  # list of field names to delete from this XBlock
+        for field_name in self.editable_fields:
+            field = self.fields[field_name]
+            if field_name in data['values']:
+                if isinstance(field, JSONField):
+                    values[field_name] = field.from_json(data['values'][field_name])
+                else:
+                    raise JsonHandlerError(400, "Unsupported field type: {}".format(field_name))
+            elif field_name in data['defaults'] and field.is_set_on(self):
+                to_reset.append(field_name)
+        self.clean_studio_edits(values)
+        validation = Validation(self.scope_ids.usage_id)
+        # We cannot set the fields on self yet, because even if validation fails, studio is going to save any changes we
+        # make. So we create a "fake" object that has all the field values we are about to set.
+        preview_data = FutureFields(
+            new_fields_dict=values,
+            newly_removed_fields=to_reset,
+            fallback_obj=self
+        )
+        self.validate_field_data(validation, preview_data)
+        if validation:
+            for field_name, value in values.iteritems():
+                setattr(self, field_name, value)
+            for field_name in to_reset:
+                self.fields[field_name].delete_from(self)
+            return {'result': 'success'}
+        else:
+            raise JsonHandlerError(400, validation.to_json())
+
+    def clean_studio_edits(self, data):
+        """
+        Given POST data dictionary 'data', clean the data before validating it.
+        e.g. fix capitalization, remove trailing spaces, etc.
+        """
+        # Example:
+        # if "name" in data:
+        #     data["name"] = data["name"].strip()
+        pass
+
+    def validate_field_data(self, validation, data):
+        """
+        Validate this block's field data. Instead of checking fields like self.name, check the
+        fields set on data, e.g. data.name. This allows the same validation method to be re-used
+        for the studio editor. Any errors found should be added to "validation".
+
+        This method should not return any value or raise any exceptions.
+        All of this XBlock's fields should be found in "data", even if they aren't being changed
+        or aren't even set (i.e. are defaults).
+        """
+        # Example:
+        # if data.count <=0:
+        #     validation.add(ValidationMessage(ValidationMessage.ERROR, u"Invalid count"))
+        pass
```

Comment:
Same thing - some subclasses may not have any validation to do, or may have more complex validation requirements that require them to overload `validate` rather than this more simplified method.